package/kernel/mac80211/files/lib/wifi/mac80211.sh

```diff
@@ -76,21 +76,27 @@ detect_mac80211() {
 		config_foreach check_mac80211_device wifi-device
 		[ "$found" -gt 0 ] && continue
 
-		mode_band="g"
-		channel="11"
-		htmode=""
-		ht_capab=""
-
-		iw phy "$dev" info | grep -q 'Capabilities:' && htmode=HT20
+		if ! [ iw phy "$dev" info | grep -q '58320 MHz' ]; then
```

Comment:
Don't think you need the brackets here?

Reply:
And is the negation (!) correct in this case?

Reply:
Gotta test whether brackets are needed but ! negation is incorrect, it's leftover and I hit the case on hAP ac2 where it set ad as hwmode.
Will fix that

Reply:
Looking at it again, I know why ! is used.
Its because grep -q will return 0 if match occurred, but there is also a downside that it returns 0 even if an error occurs.
So this will be changed

Reply:
Can you give an example? In a quick test it works for me.